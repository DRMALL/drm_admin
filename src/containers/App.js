

import React, { Component } from 'react'
import store from '../commons/store'
import '../styles/css/index.css'
import 'antd/dist/antd.css'



export default class App extends Component {
  constructor(props) {
    super(props)
    this.state = store.getState()
  }

  componentDidMount() {
    this.unsubscribe = store.subscribe( ()=> this.setState(store.getState()) )
    // verifyToken()
  }

  componentWillUnmount() {
    this.unsubscribe()
  }

  render() {
    return(
      <div>
        {
          this.props.children && React.cloneElement(this.props.children, { state: this.state })
        }
      </div>
      )
  }
}
