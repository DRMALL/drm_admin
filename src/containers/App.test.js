

import React from 'react'
import App from './App'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

it('App test', () => {
  const wrapper = render(<App />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
