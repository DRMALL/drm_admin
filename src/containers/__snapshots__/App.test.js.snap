// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test 1`] = `<div />`;
